test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+                      , tid);
+        }
+    }
+    else
+    {
+        LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- the return value is an unexpected one", tid);
+    }
+}
+
+void check_story_destroyed(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+
+    if(ret == chronolog::CL_SUCCESS)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_RELEASED)
+        {
+            shared_state[tid] = static_cast<int>(ThreadState::STORY_DESTROYED);
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- State updated to: {}", tid, get_state_name(
+                    ThreadState::STORY_DESTROYED));
+            update_story_threads_state(tid, ThreadState::STORY_DESTROYED);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_SUCCESS return value when"
+                      " trying to destroy a story on a state different from STORY_RELEASED", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_ACQUIRED)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_ACQUIRED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_ACQUIRED return value when "
+                     "trying to destroy a Story on the STORY_ACQUIRED state.", tid);
+        }
+        else
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_ACQUIRED return value when "
+                     "trying to destroy a story on a state different from STORY_ACQUIRED state.", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_NOT_EXIST)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::CHRONICLE_DESTROYED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                     " trying to destroy a Story on the STORY_DESTROYED or CHRONICLE_DESTROYED state.", tid);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                      "trying to destroy a story on a state different than STORY_DESTROYED or CHRONICLE_DESTROYED "
+                      "state.", tid);
+        }
+    }
+    else
+    {
+        LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- the return value is an unexpected one", tid);
+    }
+}
+
+void check_chronicle_destroyed(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+    if(ret == chronolog::CL_SUCCESS)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_RELEASED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED)
```

Comment:
I think the return value should be `CL_ERR_ACQUIRED` if this thread is in `STORY_DESTROYED` as a result of another thread destroyed this Story before this thread tries to destroy it.

Reply:
The story is destroyed, but the chronicle is not. If another thread destroyed the chronicle, then this thread would be in the `CHRONICLE_DESTROYED` state.